### PR TITLE
feat: add a setting for the default calendar url

### PR DESCRIPTION
**Pull request by st3iny**
Resolves https://github.com/nextcloud/calendar/issues/1835

Based on https://github.com/nextcloud/calendar/pull/2331 and https://github.com/nextcloud/calendar/pull/4310

Closes https://github.com/nextcloud/calendar/pull/2331
Closes https://github.com/nextcloud/calendar/pull/4310

Requires:
- https://github.com/nextcloud/cdav-library/pull/871
- https://github.com/nextcloud/server/pull/43745

grafik

## TODO

- [x] Use the default calendar when creating new events
- [x] Fix tests

### CODE REVIEW

File: src/components/AppNavigation/Settings.vue [hunk 1]
```diff
@@ -272,6 +290,27 @@ export default {
 		nextcloudVersion() {
 			return parseInt(OC.config.version.split('.')[0])
 		},
+		defaultCalendarOptions() {
+			return this.$store.state.calendars.calendars.filter(calendar => !calendar.readOnly)
```
Comment (st3iny) [inline src/components/AppNavigation/Settings.vue:294]: We should now allow incoming shared calendars. The backend only allows owned calendars for now to prevent unnecessary complexity (ACLs etc.).

Comment (st3iny) [reply src/components/AppNavigation/Settings.vue]: Done

**Review by tcitworld — COMMENTED** [review]
Since it requires https://github.com/nextcloud/server/pull/43745, shouldn't this only be showed on NC 29+ ?

**Comment by st3iny** [comment]
Good catch! I'll add a version check.

**Review by ChristophWurst — APPROVED** [review]
Makes sense

**Review by hamza221 — APPROVED** [review]

**Comment by Sebastian-Roth** [comment]
Thanks to everyone involved for adding this feature. I just started to use NectCloud Calendar and have exact that situation where I don't want to have the personal calendar as default.

Unfortunately the setting does not work as expected. Not sure if this is the right place to report this but I will give it a go. We use a shared calendar as a group. The calendar was created by the NextCloud admin user and made available to a group of people (group defined in our NextCloud). People see the shared calendar and add events but don't seem to be able to set it as default calendar - see screenshots below. Admin is able to set this shared calendar as default but still is not used on new events.

NextCloud Server version: 30.0.2
NextCloud Calendar version: 5.0.8

1.  Issue - default calendar dropdown missing for shared calendar:
user-setting-not-available

2. Issue - default calendar set but new events still use the personal calendar as default:
admin-setting-available-but-not-working

**Comment by st3iny** [comment]
@Sebastian-Roth Thanks for the report. Yes, this is working as intended. Please refer to https://github.com/nextcloud/calendar/pull/5996 and https://github.com/nextcloud/calendar/issues/5942 for some more information.

TL;DR: Some users were confused by the changed default option in the calendar picker as there are some restrictions on which calendars can be picked as the default calendar picker vs in the event calendar picker.

Perhaps someone could investigate if can revert the revert now as the support for shared calendars in the backend has been improved since.

A note for the future: Please open a new ticket and reference related pull requests and tickets there. Posting in PRs pings a lot of (unrelated) people and sometimes is lost as the pull request is already closed and some people might skip the notification. Thanks :)
